File: ios/App/View/Users/UsersListView.swift
import SwiftUI

struct UsersListView: View {
  @EnvironmentObject private var page: UsersPageComponent
  @Environment(\.uiThemeTokens) private var theme

  var body: some View {
    ScrollView {
      VStack(alignment: .leading, spacing: 12) {
        searchField(
          placeholder: "Find user...",
          text: Binding(
            get: { page.searchQuery },
            set: { page.setSearchQuery($0) }
          )
        )

        if page.rows.isEmpty, !page.pager.loading {
          Text("No users found.")
            .font(.system(size: theme.typography.sizes.sm))
            .foregroundStyle(theme.colors.textSecondary)
            .frame(maxWidth: .infinity, alignment: .center)
            .padding(.vertical, 32)
        }

        LazyVStack(spacing: 12) {
          ForEach(page.rows, id: \.id) { user in
            userCard(user)
          }
        }
        if page.pager.hasNextPage {
          loadMoreButton
        }
      }
      .padding(16)
      .frame(maxWidth: .infinity, alignment: .topLeading)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
    .background(theme.colors.appBg)
    .task {
      await page.refresh(withLoading: true)
    }
    .refreshable {
      await page.refresh(withLoading: false)
    }
  }

  @ViewBuilder
  private func userCard(_ user: UserRowData) -> some View {
    HStack(spacing: 12) {
      Circle()
        .fill(theme.colors.surfaceAlt)
        .frame(width: 38, height: 38)
        .overlay(
          Text(initials(of: user.displayName))
            .font(.system(size: theme.typography.sizes.sm, weight: .semibold))
            .foregroundStyle(theme.colors.textSecondary)
        )

      VStack(alignment: .leading, spacing: 3) {
        Text(user.displayName)
          .font(.system(size: theme.typography.sizes.md, weight: .semibold))
          .foregroundStyle(theme.colors.textPrimary)
          .lineLimit(1)
        Text(user.email)
          .font(.system(size: theme.typography.sizes.xs))
          .foregroundStyle(theme.colors.textSecondary)
          .lineLimit(1)
      }

      Spacer(minLength: 0)

      if user.status.value == .disabled {
        Text("Disabled")
          .font(.system(size: theme.typography.sizes.xs, weight: .medium))
          .foregroundStyle(theme.colors.statusWarning)
      }
    }
    .padding(12)
    .frame(maxWidth: .infinity, alignment: .leading)
    .background(
      RoundedRectangle(cornerRadius: theme.radius.md, style: .continuous)
        .fill(theme.colors.surface)
        .overlay(
          RoundedRectangle(cornerRadius: theme.radius.md, style: .continuous)
            .stroke(theme.colors.border, lineWidth: 1)
        )
    )
  }

  private func initials(of displayName: String) -> String {
    let parts = displayName
      .split(separator: " ")
      .prefix(2)
      .compactMap { $0.first.map(String.init) }
    return parts.isEmpty ? "?" : parts.joined().uppercased()
  }

  @ViewBuilder
  private var loadMoreButton: some View {
    Button("Load more users") {
      Task {
        await page.loadNextPage()
      }
    }
    .buttonStyle(.plain)
    .font(.system(size: theme.typography.sizes.sm, weight: .semibold))
    .foregroundStyle(theme.colors.accent)
    .padding(.vertical, 8)
  }

  @ViewBuilder
  private func searchField(placeholder: String, text: Binding<String>) -> some View {
    TextField(placeholder, text: text)
      .textInputAutocapitalization(.never)
      .autocorrectionDisabled()
      .padding(.horizontal, 12)
      .padding(.vertical, 10)
      .font(.system(size: theme.typography.sizes.sm))
      .foregroundStyle(theme.colors.textPrimary)
      .background(
        RoundedRectangle(cornerRadius: theme.radius.md, style: .continuous)
          .fill(theme.colors.surface)
          .overlay(
            RoundedRectangle(cornerRadius: theme.radius.md, style: .continuous)
              .stroke(theme.colors.border, lineWidth: 1)
          )
      )
      .frame(maxWidth: .infinity)
  }
}

File: android/app/src/main/kotlin/com/baseapp/android/view/users/UsersListView.kt
package com.baseapp.android.view.users

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.baseapp.android.components.components
import com.baseapp.android.graphql.UserRowData
import com.baseapp.android.graphql.UserStatusValue
import com.baseapp.android.view.kit.SearchField
import com.baseapp.android.view.theme.LocalUiTheme
import kotlinx.coroutines.launch

@Composable
fun UsersListView() {
    val theme = LocalUiTheme.current
    val scope = rememberCoroutineScope()
    val page = components.usersPage

    val rows by page.rows.collectAsState()
    val searchQuery by page.searchQuery.collectAsState()
    val pagerState by page.pager.state.collectAsState()

    LaunchedEffect(Unit) {
        page.refresh(withLoading = true)
    }

    LazyColumn(
        verticalArrangement = Arrangement.spacedBy(12.dp),
        contentPadding = PaddingValues(16.dp),
        modifier = Modifier
            .fillMaxSize()
            .background(theme.colors.appBg),
    ) {
        item {
            SearchField(
                placeholder = "Find user...",
                value = searchQuery,
                onValueChange = { page.setSearchQuery(it) },
            )
        }

        if (rows.isEmpty() && !pagerState.loading) {
            item {
                Text(
                    text = "No users found.",
                    color = theme.colors.textSecondary,
                    fontSize = theme.typography.sizes.sm,
                    textAlign = TextAlign.Center,
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 32.dp),
                )
            }
        }

        items(rows, key = { it.id }) { user ->
            UserCard(user)
        }

        if (pagerState.hasNextPage) {
            item {
                Text(
                    text = "Load more users",
                    color = theme.colors.accent,
                    fontSize = theme.typography.sizes.sm,
                    fontWeight = FontWeight.SemiBold,
                    textAlign = TextAlign.Center,
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable { scope.launch { page.loadNextPage() } }
                        .padding(vertical = 8.dp),
                )
            }
        }
    }
}

@Composable
private fun UserCard(user: UserRowData) {
    val theme = LocalUiTheme.current
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(theme.radius.md))
            .background(theme.colors.surface)
            .border(1.dp, theme.colors.border, RoundedCornerShape(theme.radius.md))
            .padding(12.dp),
    ) {
        Box(
            contentAlignment = Alignment.Center,
            modifier = Modifier
                .size(38.dp)
                .clip(CircleShape)
                .background(theme.colors.surfaceAlt),
        ) {
            Text(
                text = initialsOf(user.displayName),
                color = theme.colors.textSecondary,
                fontSize = theme.typography.sizes.sm,
                fontWeight = FontWeight.SemiBold,
            )
        }

        Column(
            verticalArrangement = Arrangement.spacedBy(3.dp),
            modifier = Modifier
                .weight(1f)
                .padding(horizontal = 12.dp),
        ) {
            Text(
                text = user.displayName,
                color = theme.colors.textPrimary,
                fontSize = theme.typography.sizes.md,
                fontWeight = FontWeight.SemiBold,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
            Text(
                text = user.email,
                color = theme.colors.textSecondary,
                fontSize = theme.typography.sizes.xs,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }

        if (user.status == UserStatusValue.DISABLED) {
            Text(
                text = "Disabled",
                color = theme.colors.statusWarning,
                fontSize = theme.typography.sizes.xs,
                fontWeight = FontWeight.Medium,
            )
        }
    }
}

private fun initialsOf(displayName: String): String {
    val parts = displayName
        .split(" ")
        .filter { it.isNotEmpty() }
        .take(2)
        .mapNotNull { it.firstOrNull()?.toString() }
    return if (parts.isEmpty()) "?" else parts.joinToString("").uppercase()
}
